app/prism/Prism.scala

```diff
@@ -37,11 +38,23 @@ object Prism {
     } yield instances
   }
 
-  def instancesWithAmis(stackStageApp: SSA)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[(Instance, Option[AMI])]] = {
+  private def instancesAndAmis(stackStageApp: SSA)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[(List[Instance], List[AMI])] = {
     for {
       prodInstances <- getInstances(stackStageApp)
       amiAttempts = amiArns(prodInstances).map(getAMI)
       amis <- Attempt.successfulAttempts(amiAttempts)
-    } yield instanceAmis(prodInstances, amis)
+    } yield (prodInstances, amis)
+  }
+
+  def instancesWithAmis(stackStageApp: SSA)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[(Instance, Option[AMI])]] = {
```

Comment:
I am not sure I understand what you mean here.
`instancesWithAmis` was already in this file, requiring an execution context and a config.
The logic that associate AMIs with instances or instances with AMIs (`instanceAmis` and `amiInstances`) still live in `PrismLogic`